### DISCUSSION TITLE
build issue

### DISCUSSION THREAD

**Discussion by NaokiShikata** [souffle-lang/souffle#2246] [Q&A] [answered]
Hi, I got stuck on the installation when `cmake --build build`. I have no idea why line 313 says error. Any advice would be very appreciated. Thank you for your support.

--
/home/ubuntu/souffle/src/FunctorOps.cpp: In constructor ‘souffle::{anonymous}::FUNCTOR_INTRINSIC_SANCHECKER::FUNCTOR_INTRINSIC_SANCHECKER()’:
/home/ubuntu/souffle/src/FunctorOps.cpp:313:27: error: unused variable ‘_’ [-Werror=unused-variable]
         for (auto&& [_, xs] : byOp) {
                                      ^
cc1plus: all warnings being treated as errors
src/CMakeFiles/libsouffle.dir/build.make:86: recipe for target 'src/CMakeFiles/libsouffle.dir/FunctorOps.cpp.o' failed
make[2]: *** [src/CMakeFiles/libsouffle.dir/FunctorOps.cpp.o] Error 1
CMakeFiles/Makefile2:1249: recipe for target 'src/CMakeFiles/libsouffle.dir/all' failed
make[1]: *** [src/CMakeFiles/libsouffle.dir/all] Error 2
Makefile:165: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by b-scholz** [comment, marked as answer]
On which system/C++ compiler are you building souffle? 

Strict warning messages stop the build. You need to disable the warning message using the flag -DCMAKE_CXX_FLAGS=-werror=unused-variable

**Reply by NaokiShikata** [reply]
Thank you for addressing my issue. This is resolved.
